service/src/main/java/com/myhome/services/springdatajpa/UserSDJpaService.java

```diff
@@ -44,15 +44,12 @@
   private final UserMapper userMapper;
   private final PasswordEncoder passwordEncoder;
 
-  @Override public Optional<UserDto> createUser(UserDto request) {
-    if (userRepository.findByEmail(request.getEmail()) == null) {
-      generateUniqueUserId(request);
-      encryptUserPassword(request);
-      return Optional.of(createUserInRepository(request));
-    } else {
-      return Optional.empty();
+  @Override public void createUser(UserDto dto) {
+    verifyEmailUniqueness(dto);
```

Comment:
Let's wrap it in optional and use filter/orElse rather than putting that in void methods